portal/models/login.py

```diff
@@ -20,6 +23,39 @@ def login_user(user, auth_method=None):
     authentication method.
 
     """
+    from .message import EmailMessage  # prevent cycle
+    # beyond patients and care givers, 2FA is required.  confirm or initiate
+    if (
+            not current_app.testing and
+            user.has_role(
+                    ROLE.ADMIN.value,
+                    ROLE.ANALYST.value,
+                    ROLE.CLINICIAN.value,
+                    ROLE.PRIMARY_INVESTIGATOR.value,
+                    ROLE.CLINICIAN.value,
+                    ROLE.RESEARCHER.value,
+                    ROLE.STAFF.value,
+                    ROLE.STAFF_ADMIN.value,
+            ) and
+            session.get('2FA_verified') != '2FA verified'):
+        # log user back out, in case a flow already promoted them
+        flask_user_logout()
+
+        token = user.generate_otp()
+        session['user_needing_2fa'] = user.id
+        session['pending_auth_method'] = auth_method
+        current_app.logger.debug(f"OTP for {user.id}: {token}")
+        email = EmailMessage(
+            subject="TrueNTH Access Token",
+            body=f"One Time Authentication Token: {token}",
```

Comment:
I think we need i18n for staff, but I'm not 100% sure. @mcjustin ?

Reply:
Yes, we'll need i18n here. Current emails to staff are indeed i18n'd (as is the content on the website for them).